api/rest/client/methods_test.go

```diff
@@ -165,20 +165,59 @@ func TestPinPath(t *testing.T) {
 	api := testAPI(t)
 	defer shutdown(api)
 
+	opts := types.PinOptionsWithPath{
+		PinOptions: types.PinOptions{
+			ReplicationFactorMin: 6,
+			ReplicationFactorMax: 7,
+			Name:                 "hello there",
+		},
+	}
+
+	testCases := []struct {
+		name    string
+		path    string
+		wantErr bool
+	}{
+		// Uncomment after removing logic Pin and Unpin with cid
+		// {
+		// 	"cid string as path",
+		// 	test.TestCid5,
+		// 	false,
+		// },
+		{
+			"IPFS path with just a cid string, but with starting with /ipfs/",
+			test.TestPathIPFS1,
```

Comment:
> A path with double slashes and maybe a slash in the end (to check the client follows the redirects correctly).

no need to test this since gorilla doesnt follow redirect, right(https://github.com/ipfs/ipfs-cluster/pull/634#discussion_r253298728)?

Reply:
You can test them if you do path-cleanup in the client, so to see they don't error.